src/lib.rs

```diff
@@ -101,6 +102,30 @@ struct OutputAssertion {
     fuzzy: bool,
 }
 
+#[derive(Debug, Copy, Clone)]
+enum OutputType {
+    StdOut,
+    StdErr,
+}
+
+impl OutputType {
+    fn select<'a>(&self, o: &'a Output) -> &'a [u8] {
+        match *self {
+            OutputType::StdOut => &o.stdout,
+            OutputType::StdErr => &o.stderr,
+        }
+    }
+}
+
+impl fmt::Display for OutputType {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            OutputType::StdOut => write!(f, "stdout"),
+            OutputType::StdErr => write!(f, "stderr"),
+        }
+    }
+}
```

Comment:
I generally tend to prefer traits too. Hm, maybe we keep that idea in mind for the 'general architecture' discussion?